src/main/java/htmlpublisher/HtmlPublisher.java

```diff
@@ -237,6 +239,7 @@ public static boolean publishReports(Run<?, ?> build, FilePath workspace, TaskLi
         for (int i=0; i < reportTargets.size(); i++) {
             // Create an array of lines we will eventually write out, initially the header.
             List<String> reportLines = new ArrayList<>(headerLines);
+            reportLines.add("<script type=\"text/javascript\" src=\"" + getStaticResourcesUrl() + "/plugin/htmlpublisher/js/htmlpublisher.js\"></script>");
```

Comment:
There's at least one flaw with the current approach that I'd like to make explicit.

We're including static resource URL prefix (`/static/XXXXXXXX`) which is the recommended way of including resources. The reference is hardcoded in the report. The issue is that prefix changes every time Jenkins is restarted. I am still able to access resources under "old" prefixes, but I guess those should expire eventually. Stapler seems to set the Expire header to [365 days](https://github.com/jenkinsci/stapler/blob/92458dd7afca3061956ccf92fedf1782a6e76e39/core/src/main/java/org/kohsuke/stapler/Stapler.java#L239), but I don't know if that's good enough to leave this as is.
I found [here](https://wiki.jenkins.io/display/JENKINS/Hyperlinks-in-HTML.html) that while not recommended resources can be accessed under the same path but without the prefix.

I'm not sure what would be better in this case.